File: 2500. Delete Greatest Value in Each Row/deleteGreatestValue.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
    public:
        int deleteGreatestValue(vector<vector<int>>& grid) {
            int ans = 0;
    
            for(auto& row : grid){
                sort(row.begin(), row.end());
            }
    
            while(grid[0].size() > 0){
                int maximum = 0;
    
                for(auto& row : grid){
                    maximum = max(maximum, row.back());
                    row.pop_back();
                }
                ans += maximum;
            }
    
            return ans;
        }
    };

File: 2500. Delete Greatest Value in Each Row/deleteGreatestValue.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var deleteGreatestValue = function (grid) {
    let ans = 0;
    for (let row of grid) {
        row.sort((a, b) => a - b)
    }

    while (grid[0].length > 0) {
        let max = 0

        for (let row of grid) {
            max = Math.max(max, row.pop())
        }
        ans += max
    }

    return ans
};